test/unit/CLI-test.js

```diff
@@ -284,77 +284,6 @@ describe('CLI class', () => {
     });
   });
 
-
-  describe('when configuration was saved', () => {
```

Comment:
This test is supposed to verify that CLI arguments can override dredd.yml arguments. I don't think this is a valuable test, as it:

- tries to unit test an integration point, at the cost of stubbing and mocking everything (see my other comment)
- tests configuration / CLI behavior using the `Dredd` class, which isn't really related to the subject under test much

I think this should be integration-tested together with revamping the configuration mechanics, not unit-tested here and definitely not tested this way.